dotnet/src/Connectors/Connectors.Qdrant.UnitTests/QdrantVectorRecordStoreTests.cs

```diff
@@ -0,0 +1,731 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Memory;
+using Moq;
+using Qdrant.Client.Grpc;
+using Xunit;
+
+namespace Microsoft.SemanticKernel.Connectors.Qdrant.UnitTests;
+
+/// <summary>
+/// Contains tests for the <see cref="QdrantVectorRecordStore{TRecord}"/> class.
+/// </summary>
+public class QdrantVectorRecordStoreTests
+{
+    private const string TestCollectionName = "testcollection";
+    private const ulong UlongTestRecordKey1 = 1;
+    private const ulong UlongTestRecordKey2 = 2;
+    private static readonly Guid s_guidTestRecordKey1 = Guid.Parse("11111111-1111-1111-1111-111111111111");
+    private static readonly Guid s_guidTestRecordKey2 = Guid.Parse("22222222-2222-2222-2222-222222222222");
+
+    private readonly Mock<MockableQdrantClient> _qdrantClientMock;
+
+    private readonly CancellationToken _testCancellationToken = new(false);
+
+    public QdrantVectorRecordStoreTests()
+    {
+        this._qdrantClientMock = new Mock<MockableQdrantClient>(MockBehavior.Strict);
+    }
+
+    [Theory]
+    [MemberData(nameof(TestOptions))]
+    public Task CanGetRecordWithVectorsAsync(bool useDefinition, bool passCollectionToMethod, bool hasNamedVectors, Type keyType)
+    {
+        if (keyType == typeof(ulong))
+        {
+            return this.CanGetRecordWithVectorsInternalAsync(useDefinition, passCollectionToMethod, hasNamedVectors, UlongTestRecordKey1);
+        }
+
+        if (keyType == typeof(Guid))
+        {
+            return this.CanGetRecordWithVectorsInternalAsync(useDefinition, passCollectionToMethod, hasNamedVectors, s_guidTestRecordKey1);
+        }
+
+        Assert.Fail("No valid key type provided.");
+        return Task.CompletedTask;
+    }
+
+    private async Task CanGetRecordWithVectorsInternalAsync<TKey>(bool useDefinition, bool passCollectionToMethod, bool hasNamedVectors, TKey testRecordKey)
```

Comment:
nit: I usually prefer to keep private methods in tests at the end, so when you open test class, you can see all test scenarios first and then all the helper methods below. But it's not critical.

Reply:
I agree, but these are not actually helper methods.  They are really the main test implementations.
That said, after having looked at it again this morning, I found a way to get rid of the separate routing method and implementation method, which was the goal all along 😄